62.CAD/fragment_shader.hlsl

```diff
@@ -15,6 +16,68 @@ void endInvocationInterlockEXT();
 // Write a general one, and maybe another one that uses precomputed values, and move these to somewhere nice in our builtin hlsl shaders if we don't have one
 // See: https://github.com/erich666/GraphicsGems/blob/master/gems/Roots3And4.c
 
+template<typename float_t, typename CurveType>
+struct LineStyleClipper

```

Comment:
@devshgraphicsprogramming

Reply:
`CurveType` could also have a `using scalar_t = float_t` or `using float_t = FloatType` alias inside it